builtin/credential/aws/path_role.go

```diff
@@ -348,14 +399,127 @@ func (b *backend) pathRoleCreateUpdate(
 		roleEntry.BoundIamInstanceProfileARN = boundIamInstanceProfileARNRaw.(string)
 	}
 
-	// Ensure that at least one bound is set on the role
-	switch {
-	case roleEntry.BoundAccountID != "":
-	case roleEntry.BoundAmiID != "":
-	case roleEntry.BoundIamInstanceProfileARN != "":
-	case roleEntry.BoundIamRoleARN != "":
-	default:
+	if boundIamPrincipalARNRaw, ok := data.GetOk("bound_iam_principal_arn"); ok {
+		roleEntry.BoundIamPrincipalARN = boundIamPrincipalARNRaw.(string)
+	}
+
+	if inferRoleTypeRaw, ok := data.GetOk("role_inferred_type"); ok {
+		roleEntry.RoleInferredType = inferRoleTypeRaw.(string)
+	}
+
+	if inferredAWSRegionRaw, ok := data.GetOk("inferred_aws_region"); ok {
+		roleEntry.InferredAWSRegion = inferredAWSRegionRaw.(string)
+	}
+
+	var allowEc2Auth, allowIamAuth bool
+
+	parseAllowedAuthTypes := func(input string) string {
```

Comment:
Its very confusing as to what the return type is. Can we make this an error here as well?